lib/Doctrine/ORM/Id/AbstractIdGenerator.php

```diff
@@ -19,19 +19,19 @@
 
 namespace Doctrine\ORM\Id;
 
-use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\EntityManagerInterface;
 
 abstract class AbstractIdGenerator
 {
     /**
      * Generates an identifier for an entity.
      *
-     * @param \Doctrine\ORM\EntityManager  $em
+     * @param EntityManagerInterface       $em
      * @param \Doctrine\ORM\Mapping\Entity $entity
      *
      * @return mixed
      */
-    abstract public function generate(EntityManager $em, $entity);
+    abstract public function generate(EntityManagerInterface $em, $entity);
```

Comment:
Not actually a BC break, as it is in ORM internals, but I will have to document the change